src/audio/drc/drc_math_generic.c

```diff
@@ -243,5 +217,39 @@ inline int32_t drc_inv_fixed(int32_t x, int32_t precision_x, int32_t precision_y
 #undef qc
 }
 
+int32_t drc_mult_lshift(int32_t a, int32_t b, int32_t lshift) { return 0; }
+int32_t drc_get_lshift(int32_t qa, int32_t qb, int32_t qy) { return 0; }
+
+#endif /* DRC_GENERIC */
+
+/*
+ * Input is Q2.30: (-2.0, 2.0)
+ * Output range: (-1.0, 1.0); regulated to Q1.31: (-1.0, 1.0)
+ */
+inline int32_t drc_sin_fixed(int32_t x)
+{
+	const int32_t PI_OVER_TWO = Q_CONVERT_FLOAT(1.57079632679489661923f, 30);
+
+	/* input range of sin_fixed() is non-negative */
+	int32_t abs_sin_val = sin_fixed(q_mult(ABS(x), PI_OVER_TWO, 30, 30, 28));
+
+	return SGN(x) < 0 ? -abs_sin_val : abs_sin_val;
+}
+
+/*
+ * Input x is Q6.26; valid range: (0.0, 32.0); x <= 0 is not supported
+ *       y is Q2.30: (-2.0, 2.0)
+ * Output is Q12.20: max 2048.0
+ */
+inline int32_t drc_pow_fixed(int32_t x, int32_t y)
```

Comment:
we should move generic math functions like this to the math lib, so we don't reimplement ones that other components might be using

Reply:
We could have further discussion with Intel to move these functions and do it in the future PR.

Reply:
sgtm